packages/core/src/utils.ts

```diff
@@ -331,3 +331,38 @@ export class RateLimitter {
     await await Promise.race(this.processingQueue.values())
   }
 }
+
+export function mergeQueries<T extends Doc> (query1: DocumentQuery<T>, query2: DocumentQuery<T>): DocumentQuery<T> {
+  const q = { ...query1 }
+  for (const k in query2) {
+    if (
+      typeof query2[k] === 'object' &&
+      typeof query1[k] === 'object' &&
+      Object.keys(query2[k]).every((t) => t.startsWith('$')) &&
```

Comment:
query1[k] could be non object and we still should merge.
Please use isPredicate(query2[key])
query2[k] could also be non object and we should merge.

```ts
query1 = {
    space: prj1._id
}
query2 = {
   space: {$in: [prj1._id, prj2._id]
}
```
and wise verse